triggering.go

```diff
@@ -35,22 +35,22 @@ type TriggerState struct {
 }
 
 // create a record using dsp.NPresamples and dsp.NSamples
-func (dsp *DataStreamProcessor) triggerAt(segment *DataSegment, i int) *DataRecord {
-	record := dsp.triggerAtSpecificSamples(segment, i, dsp.NPresamples, dsp.NSamples)
+func (dsp *DataStreamProcessor) triggerAt(stream *DataStream, i int) *DataRecord {
```

Comment:
I guess we can do this. But what's the pedantic argument?

Reply:
it's to be conceptually consistent. The DSP is a `DataStreamProcessor`. It makes sense that it processes a `DataStream`. It's confusing when it sometimes handles a `DataStream` and other time handles a `DataSegment`. One spends mental effort trying to figure out why one function takes one and another takes the other. This way they all just take `DataStream` and you don't need to wonder why you would choose `Stream` or `Segment` in any part of the code.

Reply:
This makes sense.